XMonad/Util/NamedScratchpad.hs

```diff
@@ -219,14 +226,16 @@ nsHideOnFocusLoss scratches = withWindowSet $ \winSet -> do
 
 -- | Execute some action on a named scratchpad.
 --
--- NOTE: Ignores its first argument and uses 'NSPState' instead.
+-- NOTE: Most of the time, this function ignores its first argument and
+-- uses 'NSPState' instead.  The only time when it does not is when no
+-- other window has been opened before in the running xmonad instance.
```

Comment:
Those other `NOTE`s for functions like `spawnHereNamedScratchpadAction` and `allNamedScratchpadAction` etc. are now outdated as well. Maybe just refer to the note here instead of copypasting?

Reply:
I liked the ghc-style Note we had recently. Should we consider that for future notes as well?

Reply:
That one wasn't rendered by haddock (iiuic, which I probably don't) while this one needs to be. But if it can be made to work (or perhaps even just works), why not.

Reply:
That has nothing to do with the note style, it's just using the right markup to start the comment block.

Reply:
To be more clear, this is rendered by the `-- |` and is where the new-style note would go. The others would then have `See Note …`.

Reply:
I think it's neat having something to refer to by name, so I've given the user-facing note a name for now